mono/metadata/jit-info.c

```diff
@@ -118,7 +121,10 @@ jit_info_table_free (MonoJitInfoTable *table, gboolean duplicate)
 		MonoJitInfoTableChunk *chunk = table->chunks [i];
 		MonoJitInfo *tombstone;
 
-		if (--chunk->refcount > 0)
+		mono_atomic_dec_i32 (&chunk->refcount);
```

Comment:
Same.

Reply:
I was seeing chunks that the call in the signal handler observed to be full of garbage. I placed assertions after each write. When it didn't hit any assertions that massive values were being written into pointers, I threw jemalloc at it with the zero-on-alloc, fill-with-canaries-on-free mode enabled, and the garbage memory was full of canaries. 

That meant that we were reading it after it was freed. (That this sees a 0 refcount but that another thread expected a non-zero refcount). Adding this code caused this bug to no longer reproduce.